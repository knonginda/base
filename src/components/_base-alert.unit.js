/* eslint-disable no-undef */
import BaseAlert from './_base-alert'

describe('@components/_base-alert', () => {
  it('renders its content', () => {
    const slotTitleContent = '<span>title</span>'
    const slotContent = '<span>foo</span>'
    const wrapper = shallowMount(BaseAlert, {
      slots: {
        title: slotTitleContent,
        default: slotContent,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders its content when theme equals success', () => {
    const slotTitleContent = '<span>title</span>'
    const slotContent = '<span>foo</span>'
    const wrapper = shallowMount(BaseAlert, {
      propsData: {
        theme: 'success',
      },
      slots: {
        title: slotTitleContent,
        default: slotContent,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders its content when theme equals warning', () => {
    const slotTitleContent = '<span>title</span>'
    const slotContent = '<span>foo</span>'
    const wrapper = shallowMount(BaseAlert, {
      propsData: {
        theme: 'warning',
      },
      slots: {
        title: slotTitleContent,
        default: slotContent,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders its content when theme equals danger', () => {
    const slotTitleContent = '<span>title</span>'
    const slotContent = '<span>foo</span>'
    const wrapper = shallowMount(BaseAlert, {
      propsData: {
        theme: 'danger',
      },
      slots: {
        title: slotTitleContent,
        default: slotContent,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders its content when theme equals info', () => {
    const slotTitleContent = '<span>title</span>'
    const slotContent = '<span>foo</span>'
    const wrapper = shallowMount(BaseAlert, {
      propsData: {
        theme: 'info',
      },
      slots: {
        title: slotTitleContent,
        default: slotContent,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
