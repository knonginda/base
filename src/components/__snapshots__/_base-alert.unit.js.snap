// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/_base-alert renders its content 1`] = `
<div class="baseAlert baseAlertDanger">
  <basefasicon-stub source="font-awesome" name="times-circle" class="themeIcon"></basefasicon-stub>
  <div class="msgContainer">
    <div class="baseTitle"><span>title</span></div> <span>foo</span>
  </div>
</div>
`;

exports[`@components/_base-alert renders its content when theme equals danger 1`] = `
<div class="baseAlert baseAlertDanger">
  <basefasicon-stub source="font-awesome" name="times-circle" class="themeIcon"></basefasicon-stub>
  <div class="msgContainer">
    <div class="baseTitle"><span>title</span></div> <span>foo</span>
  </div>
</div>
`;

exports[`@components/_base-alert renders its content when theme equals info 1`] = `
<div class="baseAlert baseAlertInfo">
  <basefasicon-stub source="font-awesome" name="exclamation-circle" class="themeIcon"></basefasicon-stub>
  <div class="msgContainer">
    <div class="baseTitle"><span>title</span></div> <span>foo</span>
  </div>
</div>
`;

exports[`@components/_base-alert renders its content when theme equals success 1`] = `
<div class="baseAlert baseAlertSuccess">
  <basefasicon-stub source="font-awesome" name="check-circle" class="themeIcon"></basefasicon-stub>
  <div class="msgContainer">
    <div class="baseTitle"><span>title</span></div> <span>foo</span>
  </div>
</div>
`;

exports[`@components/_base-alert renders its content when theme equals warning 1`] = `
<div class="baseAlert baseAlertWarning">
  <basefasicon-stub source="font-awesome" name="exclamation-triangle" class="themeIcon"></basefasicon-stub>
  <div class="msgContainer">
    <div class="baseTitle"><span>title</span></div> <span>foo</span>
  </div>
</div>
`;
